admin/class-gutenberg-compatibility.php

```diff
@@ -0,0 +1,132 @@
+		$this->current_version = $this->detect_installed_gutenberg_version();
+	}
+
+	/**
+	 * Determines whether or not Gutenberg is installed.
+	 *
+	 * @return bool Whether or not Gutenberg is installed.
+	 */
+	public function is_installed() {
+		return $this->is_gutenberg_available() && $this->current_version !== '';
+	}
+
+	/**
+	 * Determines whether or not the currently installed version of Gutenberg is below the minimum supported version.
+	 *
+	 * @return bool True if the currently installed version is below the minimum supported version. False otherwise.
+	 */
+	public function is_below_minimum() {
+		return $this->get_major_minor_version( $this->current_version ) < $this->get_major_minor_version( $this->get_minimum_supported_version() );
+	}
+
+	/**
+	 * Determines whether or not the currently installed version of Gutenberg is the latest known version.
+	 *
+	 * @return bool True if the currently installed version is the latest known version. False otherwise.
+	 */
+	public function is_latest_version() {
+		return $this->get_major_minor_version( $this->current_version ) === $this->get_major_minor_version( $this->get_latest_release() );
+	}
+
+	/**
+	 * Gets the currently installed version.
+	 *
+	 * @return string The currently installed version.
+	 */
+	public function get_installed_version() {
+		return $this->current_version;
+	}
+
+	/**
+	 * Determines whether or not the currently installed version of Gutenberg is the latest, fully compatible version.
+	 *
+	 * @return bool Whether or not the currently installed version is fully compatible.
+	 */
+	public function is_fully_compatible() {
+		return ! $this->is_below_minimum() && $this->is_latest_version();
+	}
+
+	/**
+	 * Gets the latest released version of Gutenberg.
+	 *
+	 * @return string The latest release.
+	 */
+	public function get_latest_release() {
+		return self::CURRENT_RELEASE;
+	}
+
+	/**
+	 * Gets the minimum supported version of Gutenberg.
+	 *
+	 * @return string The minumum supported release.
+	 */
+	public function get_minimum_supported_version() {
+		return self::MINIMUM_SUPPORTED;
+	}
+
+	/**
+	 * Determines whether or not Gutenberg is available.
+	 *
+	 * @return bool Whether or not Gutenberg is available.
+	 */
+	public function is_gutenberg_available() {
+		return defined( 'GUTENBERG_VERSION' );
+	}
+
+	/**
+	 * Gets the major/minor version of the plugin for easier comparing.
+	 *
+	 * @param string $version The version to trim.
+	 *
+	 * @return string The major/minor version of the plugin.
+	 */
+	protected function get_major_minor_version( $version ) {
+		$version_numbers = explode( '.', $version );
+
+		return implode( '.', array_slice( $version_numbers, 0, 2 ) );
+	}
+
+	/**
+	 * Detects the currently installed Gutenberg version.
+	 *
+	 * @return string The currently installed Gutenberg version. Empty if the version couldn't be detected.
+	 */
+	protected function detect_installed_gutenberg_version() {
+		if ( $this->is_gutenberg_available() ) {
```

Comment:
I would change  by using the content of `is_gutenberg_available` and 

```php
if ( defined( 'GUTENBERG_VERSION' ) ) {
   return GUTENBERG_VERSION;
}
```

`is_gutenberg_available` can just contain the implementation of this function:
```php
    return $this->detect_installed_gutenberg_version !== '' );
```